### ISSUE TITLE
False positive / Unexpected behavior for Language level migration -> 'get_class usage correctness'

### ISSUE THREAD

**Issue by tfettig01** [kalessil/phpinspectionsea#784]
See below code:

```php
<?php

class Bar
{
    /**
     * @return Bar|null
     */
    public static function getBar(): ?Bar
    {
        return null;
    }

    /**
     * @param Bar $Bar
     */
    public static function setBar(Bar $Bar): void
    {
        $blah = $Bar;
    }
}

$Bar = Bar::getBar();
if ($Bar) {
    // False positive here.
    get_class($Bar);
    
    // Expected behavior here.
    Bar::setBar($Bar);
}
```

Because of the if check I would expect to see no inspection error on get_class() the same as there is no inspection error on Bar::setBar().

**Comment by kalessil** [comment]
It's a known issues: https://youtrack.jetbrains.com/issue/WI-38622

The `Null reference` inspection is a very intelligent beast, with own (cpu greedy) control flow analysis, hence it doesn't suffer from the issue.

**Comment by kalessil** [comment]
Also, if the case gets reported again, I'll take care of a workaround.

**Comment by kalessil** [comment]
Damn, we had similar reports already: #747, time for a workaround...

**Comment by kalessil** [comment]
Fixed
